src/chainstate/stacks/events.rs

```diff
@@ -28,12 +30,16 @@ impl TransactionOrigin {
         match self {
             TransactionOrigin::Burn(txid) => txid.clone(),
             TransactionOrigin::Stacks(tx) => tx.txid(),
+            TransactionOrigin::NetworkProtocol => Txid([0; 32]),
         }
     }
-    pub fn serialize_to_vec(&self) -> Vec<u8> {
+    /// Serialize this origin type to a string that can be stored in
+    ///  a database
+    pub fn serialize_to_dbstring(&self) -> String {
         match self {
-            TransactionOrigin::Burn(txid) => txid.as_bytes().to_vec(),
-            TransactionOrigin::Stacks(tx) => tx.txid().as_bytes().to_vec(),
+            TransactionOrigin::Burn(txid) => format!("BTC({})", txid),
+            TransactionOrigin::Stacks(tx) => to_hex(&tx.serialize_to_vec()),
+            TransactionOrigin::NetworkProtocol => "ImplicitNetworkTransaction".into(),
```

Comment:
Technically, this event isn't a transaction -- it doesn't originate from a transaction.  What if we just call this something like "bookkeeping" or "internals" without the word "transaction" so users don't get confused?